src/Tools/dotnet-monitor/CollectionRules/CollectionRulePipeline.cs

```diff
@@ -241,5 +239,14 @@ private void InvokeStartCallback()
                 _invokedStartCallback = true;
             }
         }
+
+        public CollectionRulePipelineState GetPipelineState()
+        {
+            CollectionRulePipelineState pipelineStateCopy = new CollectionRulePipelineState(_stateHolder);
+
+            pipelineStateCopy.CanExecuteActions(Context.Clock.UtcNow.UtcDateTime);
```

Comment:
Can you add some comments here? Another option is to possibly separate CanExecuteActions into two apis, one that Dequeues the timestamps and one that evaluates if an action can be executed.

Reply:
Alternatively, maybe we could use a name that's more indicative of the fact that this is responsible for updating state and not just checking a condition? I personally prefer the unification of these two tasks, because it means the caller doesn't need to remember to call dequeue - they can assume that `ExecutionTimestamps` is always up-to-date when it's operated against.